DylibBinder/DBGenericParameter.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.Collections.Generic;
+using Dynamo.SwiftLang;
+using SwiftReflector;
+using SwiftReflector.Inventory;
+
+namespace DylibBinder {
+	internal class DBGenericParameter : IEquatable<DBGenericParameter> {
+		public DBGenericParameter (int depth, int index)
+		{
+			Depth = depth;
+			Index = index;
+			Name = SLGenericReferenceType.DefaultNamer (depth, index);
+		}
+
+		public int Depth { get; }
+		public int Index { get; }
+		public string Name { get; }
+
+		public bool Equals (DBGenericParameter other)
+		{
+			if (this.Depth == other.Depth && this.Index == other.Index)
+				return true;
+			return false;
+		}
+	}
+
+	internal class DBGenericParameters {
+
+		public DBGenericParameters (SwiftBaseFunctionType signature) {
+			AssignGenericParameters (signature);
+		}
+
+		public DBGenericParameters (SwiftType swiftType)
+		{
+			FilterGenericParameterSource (swiftType);
+		}
+
+		public DBGenericParameters (DBTypeDeclaration typeDeclaration)
+		{
+			ParseTopLevelGenerics (typeDeclaration.Funcs, typeDeclaration.Properties);
+		}
+
+		public List<DBGenericParameter> GenericParameters { get; } = new List<DBGenericParameter> ();
+
+
+		void AssignGenericParameters (SwiftBaseFunctionType signature)
+		{
+			FilterGenericParameterSource (signature.Parameters, signature.ReturnType);
+		}
+
+		void FilterGenericParameterSource (params SwiftType [] types)
+		{
+			foreach (var type in types) {
+				switch (type) {
+				case SwiftFunctionType funcType:
+					HandleFunctionType (funcType);
+					return;
+				case SwiftBoundGenericType boundType:
+					HandleBoundGenericType (boundType);
+					return;
+				case SwiftTupleType tupleType:
+					HandleTupleType (tupleType);
+					return;
+				case SwiftGenericArgReferenceType refType:
+					HandleGenericArgReferenceType (refType);
+					return;
+				default:
+					return;
+				}
+			}
+		}
+
+		void HandleFunctionType (SwiftFunctionType funcType)
+		{
+			switch (funcType.Parameters) {
+			case SwiftBoundGenericType boundType:
+				HandleBoundGenericType (boundType);
+				return;
+			case SwiftTupleType tupleType:
+				HandleTupleType (tupleType);
+				return;
+			case SwiftGenericArgReferenceType refType:
+				HandleGenericArgReferenceType (refType);
+				return;
+			default:
+				return;
+			}
+		}
+
+		void HandleBoundGenericType (SwiftBoundGenericType boundType)
```

Comment:
Same, you can rename if to HandleType, parameter overloading and parameter name will make if understandable.